### PR TITLE
chore: switch to ruff

### PR DESCRIPTION
**Pull request by lengau**
- [x] Have you followed the guidelines for contributing?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox`?

-----

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -71,14 +71,14 @@ allowlist_externals =
...
-[testenv:lint-{black,ruff,shellcheck,codespell,yaml}]
+[testenv:lint-{ruff,shellcheck,codespell,yaml}]
 description = Lint the source code
 base = testenv, lint
 labels = lint
 commands_pre =
     shellcheck: bash -c '{[shellcheck]find} | {[shellcheck]filter} > {env_tmp_dir}/shellcheck_files'
 commands =
-    black: black --check --diff {tty:--color} {posargs} .
+    ruff: ruff format --diff {posargs:.}
```
Comment (tigarmo) [inline tox.ini:81]: how does this work with two `ruff: ` entries?

File: tox.ini [hunk 2]
```diff
@@ -98,12 +98,12 @@ commands =
...
     mypy: mypy --install-types --non-interactive {posargs:.}
 
-[testenv:format-{black,ruff,codespell}]
+[testenv:format-{ruff,codespell}]
 description = Automatically format source code
 base = testenv, lint
 labels = format
 commands =
-    black: black {tty:--color} {posargs} .
+    ruff: ruff format {posargs} .
```
Comment (tigarmo) [inline tox.ini:106]: here too

File: tox.ini (hunk 1 above)
Comment (dariuszd21) [inline tox.ini:81]:
```suggestion
    ruff: ruff format --check --diff {posargs:.}
```

File: tox.ini (hunk 2 above)
Comment (dariuszd21) [inline tox.ini:106]: Above you use `{posargs:.}` and here you're going with `{posargs} .`, shouldn't it be aligned ?

File: pyproject.toml [hunk 3]
```diff
@@ -220,7 +216,7 @@ lint.select = [  # Base linting rule selections.
     "RUF001", "RUF002", "RUF003",  # Ambiguous unicode characters
     "RUF005",  # Encourages unpacking rather than concatenation
     "RUF008",  # Do not use mutable default values for dataclass attributes
-    "RUF011",  # Don't use static keys in dict comprehensions.
+    "B035",   # Don't use static keys in dict comprehensions.
```
Comment (dariuszd21) [inline pyproject.toml:219]: There is `"B0",  # Common mistakes and typos.` above, maybe you don't need it at all ?

File: tox.ini (hunk 1 above)
Comment (lengau) [reply tox.ini:81]: It runs the entries in definition order.

Comment (lengau) [reply tox.ini:81]: `--diff` implies `--check`

File: pyproject.toml (hunk 3 above)
Comment (lengau) [reply pyproject.toml:219]: Good point. Removed!

File: tox.ini (hunk 2 above)
Comment (lengau) [reply tox.ini:106]: Well spotted. Adjusted

Comment (lengau) [reply tox.ini:106]: As a sidenote, `--respect-gitignore` is now default so I removed that too.

**Review by tigarmo — APPROVED** [review]

**Comment by dariuszd21** [comment]
Is it possible, that some changes haven't been pushed ? I can't find the commit that has changes that were done to address my comments :smile:

**Comment by lengau** [comment]
Sorry about that @dariuszd21 ! Yes, I forgot to push :-)

**Review by dariuszd21 — APPROVED** [review]
Looks good, thanks!

**Pull request merged**